eml/komand_eml/util/utils.py

```diff
@@ -1,88 +1,76 @@
 # Custom imports below
 import re
-import email
 from bs4 import UnicodeDammit
 
 
 def body(b, log):
     if b.is_multipart():
+        payload = None
         for part in b.walk():
-            ctype = part.get_content_type()
-            cdispo = str(part.get('Content-Disposition'))
+            content_type = part.get_content_type()
+            content_disposition = str(part.get('Content-Disposition'))
 
-            # skip any text/plain (txt) attachments
-            if ctype == 'text/plain' and 'attachment' not in cdispo:
+            if content_type == 'text/plain' and 'attachment' not in content_disposition:
                 try:
-                    body = part.get_payload(decode=True).decode('utf-8')  # decode
+                    payload = part.get_payload(decode=True).decode('utf-8')
                 except Exception as ex:
                     log.debug(ex)
                     log.debug("Failed to parse message as UTF-8, attempting to detwingle first before retrying parse")
-                    body = UnicodeDammit.detwingle(part.get_payload(decode=True)).decode('utf-8', errors='ignore')
-            elif ctype == 'text/html' and 'attachment' not in cdispo:
+                    payload = UnicodeDammit.detwingle(part.get_payload(decode=True)).decode('utf-8', errors='ignore')
+            elif content_type == 'text/html' and 'attachment' not in content_disposition:
                 try:
-                    body = part.get_payload(decode=True).decode('utf-8').replace('\n','')  # decode
+                    payload = part.get_payload(decode=True).decode('utf-8').replace('\n', '')  # decode
                 except Exception as ex:
                     log.debug(ex)
                     log.debug("Failed to parse message as UTF-8, attempting to detwingle first before retrying parse")
+                    payload = UnicodeDammit.detwingle(part.get_payload(decode=True)).decode('utf-8', errors='ignore').replace('\n', '')
 
-                    body = UnicodeDammit.detwingle(part.get_payload(decode=True)).decode('utf-8', errors='ignore').replace('\n','')
-
-
-        return body
+        return payload
     else:
         try:
             return b.get_payload(decode=True).decode('utf-8')
-        except:
+        except Exception as e:
```

Comment:
We should make this a PluginException. Want to replace all Exceptions with those.

Reply:
PluginException is what we would raise but we do not raise an exception here or anywhere else. 
On that note I would really need to dig into this to know for sure but it seems like that could lead to unhandled exceptions.

Reply:
> We should make this a PluginException. Want to replace all Exceptions with those.

We do not want to `except` on PluginException - nothing will be caught then. You want to `raise` PluginExceptions when catching Exceptions.